src/KDSoapServer/KDSoapServerSocket.cpp

```diff
@@ -141,6 +141,20 @@ static QByteArray httpResponseHeaders(bool fault, const QByteArray &contentType,
     httpResponse += QByteArray::number(responseDataSize);
     httpResponse += "\r\n";
 
+    KDSoapServerObjectInterface *serverObjectInterface = qobject_cast<KDSoapServerObjectInterface *>(serverObject);
+    if (serverObjectInterface) {
+        const KDSoapServerObjectInterface::HttpResponseHeaderItems& additionalItems = serverObjectInterface->additionalHttpResponseHeaderItems();
+        Q_FOREACH (const KDSoapServerObjectInterface::HttpResponseHeaderItem& headerItem, additionalItems) {
+            httpResponse += headerItem.first.toLatin1();
+            httpResponse += ": ";
+            httpResponse += headerItem.second.toLatin1();
+            httpResponse += "\r\n";
+        }
+    } else {
+        qWarning() << QString::fromLatin1("Server object %1 does not implement KDSoapServerObjectInterface!").arg(QString::fromLatin1(serverObject->metaObject()->className()))
```

Comment:
Since additional items is an optional feature, I would just remove this warning (i.e. the else branch) altogether.